assets/js/phoenix_live_view.js

```diff
@@ -682,9 +682,18 @@ export class View {
     this.loader.style.display = "none"
   }
 
+  resetContainerClasses(){
+    this.el.classList.remove(
+      PHX_CONNECTED_CLASS,
+      PHX_DISCONNECTED_CLASS,
+      PHX_ERROR_CLASS
+    )
+  }
```

Comment:
I think we'll have a lot less class related bugs if we just reset the state of our known classes every time we need to rather than removing the ones which may or may not be set on a per-case basis. Happy to change this though, just let me know.

Reply:
This seems clearer.  nice!  I can back out my changes in the other PR.

Reply:
Would it be better to just have a single function that clears out our classes and then sets the one that was passed in?

`setContainerClass(target) {
  this.el.classList.remove(PHX_CONNECTED_CLASS, PHX_...);
  this.el.classList.add(target);
}`

Reply:
Yes, I think it might, but it needs the ability to support multiple classes being set at once.

We can just pass through `arguments` to do that. The spread operator would be nicer but I'm unsure about this lib's browser support.

I'll make this change ASAP.

Reply:
Marking as resolved :+1: